onboarding_project/squad_pantry_app/admin.py

```diff
@@ -22,11 +25,87 @@ class OrderDishInline(admin.StackedInline):
     formset = BaseOrderDishFormset
     extra = 1
 
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
+        if db_field.name == "dish":
+            kwargs["queryset"] = Dish.objects.filter(is_available=True)
+        return super().formfield_for_foreignkey(db_field, request, **kwargs)
+
+    def get_readonly_fields(self, request, obj=None):
+        try:
+            if obj.placed_by:
```

Comment:
We are accessing the `placed_by` attribute of `order` here. Will this cause any performance issues? Do we need the entire `placed_by` user object? Will it result in an additional query and why?

Use [django-debug-toolbar](https://django-debug-toolbar.readthedocs.io/en/stable/) to see what goes on when a page is loaded. It's a very useful tool